tests/server.rs

```diff
@@ -1482,6 +1482,72 @@ async fn http_connect_new() {
     assert_eq!(s(&vec), "bar=foo");
 }
 
+#[tokio::test]
+async fn h2_connect() {
```

Comment:
it would be interesting to have a test for CONNECTs multiplexed over a single h2 connection

Reply:
I'll write additional tests, staying away from libstd channels this time 🤡

Reply:
There is now a second stream that connects twice and makes one of the stream reset while the other one does its thing successfully.